### PR TITLE
fix(notifier): disable for json args

### PR DESCRIPTION
**Pull request by tknickman**

### CODE REVIEW

**Review by chris-olszewski — APPROVED** [review]

**Comment by chris-olszewski** [comment]
Should we put the update notification via `stderr` instead of `stdout` to avoid polluting output?

**Comment by tknickman** [comment]
Good question, every update notifier that I've used pipes to stdout, but I agree this feels like something that may be a bit annoying to maintain in the future so maybe we consider stderr

File: crates/turborepo-lib/src/shim.rs [hunk 1]
```diff
@@ -74,6 +76,13 @@ impl ShimArgs {
             })
         }
     }
+
+    // returns true if any flags result in pure json output to stdout
+    pub fn has_json_flags(&self) -> bool {
+        self.remaining_turbo_args
+            .iter()
+            .any(|arg| arg == "--json" || arg == "--dry=json" || arg == "--graph")
```
Comment (nathanhammond) [inline crates/turborepo-lib/src/shim.rs:84]: There are some additional ways to invoke this:

```
"--dry=json"
"--dry-run=json"
"--dry json"
"--dry-run json"
```

I'm not 100% sure if this is a full enumeration.

**Comment by nathanhammond** [comment]
Nah it shouldn't go to `stderr`. The real answer is to have two internal writers to `stdout`; one for structured text, one for normal output.

**Pull request merged**